diff --git a/src/driver.c b/src/driver.c
index 686d919..8c8f453 100644
--- a/src/driver.c
+++ b/src/driver.c
@@ -434,12 +434,14 @@ int main (int argc, char *argv[])
 	XStuff->display = NULL;
 	XStuff->window = 0;
 
+	/*
 	memset ((void *)&sa, 0, sizeof (struct sigaction));
 	sa.sa_handler = signalHandler;
 	sigaction (SIGINT, &sa, 0);
 	sigaction (SIGPIPE, &sa, 0);
 	sigaction (SIGQUIT, &sa, 0);
 	sigaction (SIGTERM, &sa, 0);
+	*/
 
 	/*
 	 * Connect to the X server.
