diff --git a/mscore/CMakeLists.txt b/mscore/CMakeLists.txt
index ed6c17f..9155e94 100644
--- a/mscore/CMakeLists.txt
+++ b/mscore/CMakeLists.txt
@@ -698,7 +698,7 @@ else (MINGW)
            if (APPLE)
                set(PORTMIDI_LIB portmidi)
            else (APPLE)
-               set(PORTMIDI_LIB -lportmidi -lporttime) # Remove -lporttime on RPM-based systems where PortTime is part of PortMidi.
+               set(PORTMIDI_LIB -lportmidi)
            endif (APPLE)
            target_link_libraries(mscore ${PORTMIDI_LIB})
        endif (USE_PORTMIDI)
