diff --git a/src/plugins/platforms/xcb/qxcbnativeinterface.cpp b/src/plugins/platforms/xcb/qxcbnativeinterface.cpp
index 31dedd4..0c0c45c 100644
--- a/src/plugins/platforms/xcb/qxcbnativeinterface.cpp
+++ b/src/plugins/platforms/xcb/qxcbnativeinterface.cpp
@@ -209,6 +209,18 @@ void QXcbNativeInterface::clearRegion(const QWindow *qwindow, const QRect& rect)
     }
 }
 
+void QXcbNativeInterface::setParentRelativeBackPixmap(const QWindow *qwindow)
+{
+    if (const QPlatformWindow *platformWindow = qwindow->handle()) {
+        const QXcbWindow *qxwindow = static_cast<const QXcbWindow *>(platformWindow);
+        xcb_connection_t *xcb_conn = qxwindow->xcb_connection();
+
+        const quint32 mask = XCB_CW_BACK_PIXMAP;
+        const quint32 values[] = { XCB_BACK_PIXMAP_PARENT_RELATIVE };
+        Q_XCB_CALL(xcb_change_window_attributes(xcb_conn, qxwindow->xcb_window(), mask, values));
+    }
+}
+
 void *QXcbNativeInterface::nativeResourceForIntegration(const QByteArray &resourceString)
 {
     void *result = 0;
diff --git a/src/plugins/platforms/xcb/qxcbnativeinterface.h b/src/plugins/platforms/xcb/qxcbnativeinterface.h
index 330dd00..d21c712 100644
--- a/src/plugins/platforms/xcb/qxcbnativeinterface.h
+++ b/src/plugins/platforms/xcb/qxcbnativeinterface.h
@@ -107,6 +107,7 @@ public:
     Q_INVOKABLE void beep();
     Q_INVOKABLE bool systemTrayAvailable(const QScreen *screen) const;
     Q_INVOKABLE void clearRegion(const QWindow *qwindow, const QRect& rect);
+    Q_INVOKABLE void setParentRelativeBackPixmap(const QWindow *window);
     Q_INVOKABLE bool systrayVisualHasAlphaChannel();
     Q_INVOKABLE bool requestSystemTrayWindowDock(const QWindow *window);
     Q_INVOKABLE QRect systemTrayWindowGlobalGeometry(const QWindow *window);
